src/actions/profileAction/profileAction.js

```diff
@@ -0,0 +1,89 @@
+import { profileHelper } from '../../helpers';
+import { GET_USER_PROFILE__SUCCESS, GET_USER_TRAVEL_SUCCESS, UPDATE_PROFILE__SUCCESS, GET_PLACE_SUCCESS, REQUEST_FAILURE } from '../../actionTypes';
+
+export const fetchUserProfileDetails = (token, username) => {
+  const request = profileHelper(token);
+
+  return (dispatch) => request.get(`profile/${username}`)
+    .then((response) => {
+      dispatch({
+        type: GET_USER_PROFILE__SUCCESS,
+        serverData: response.data
+      });
+    })
+    .catch((error) => {
+      dispatch({
+        type: REQUEST_FAILURE,
+        serverData: error
+      });
+    });
+};
+
+export const fetchPlaces = (token) => {
+  const request = profileHelper(token);
+
+  return (dispatch) => request.get('places')
+    .then((response) => {
+      dispatch({
+        type: GET_PLACE_SUCCESS,
+        serverData: response.data
+      });
+    })
+    .catch((error) => {
+      dispatch({
+        type: REQUEST_FAILURE,
+        serverData: error
+      });
+    });
+};
+
+export const fetchUserTravels = (token) => {
+  const request = profileHelper(token);
+
+  return (dispatch) => request.get('trips')
+    .then((response) => {
+      dispatch({
+        type: GET_USER_TRAVEL_SUCCESS,
+        serverData: response.data
+      });
+    })
+    .catch((error) => {
+      dispatch({
+        type: REQUEST_FAILURE,
+        serverData: error
+      });
+    });
+};
+
+
+export const updateUserProfileDetails = (token, username, userUpdates) => {
+  const {
+    name,
+    gender,
+    birthdate,
+    preferredLanguage,
+    preferredCurrency,
+    location
+  } = userUpdates;
+  const request = profileHelper(token);
+  return (dispatch) => request.patch(`profile/${username}`, {
+    name,
+    gender,
+    birthdate,
+    preferredLanguage,
+    preferredCurrency,
+    location
+  })
+    .then((response) => {
+      dispatch({
+        type: UPDATE_PROFILE__SUCCESS,
+        serverData: response.data,
+      });
+    })
+    .catch((error) => {
+      dispatch({
+        type: REQUEST_FAILURE,
+        serverData: error
+      });
+    });
+};
```

Comment:
Can you separate these functions, you can have a folder called profile and have different actions in each files.

Reply:
Noted with thanks